rfd/0073-discover.md

```diff
@@ -0,0 +1,337 @@
+---
+authors: Roman Tkachenko (roman@goteleport.com)
+state: draft
+---
+
+# RFD 73 - Teleport Discover
+
+## Required approvers
+
+@klizhentas && @xinding33
+
+## What
+
+Proposes the `teleport discover` command that simplifies the UX for the
+first-time users who are connecting their cloud resources to Teleport.
+
+## Related RFDs
+
+- [RFD 38: AWS databases auto-discovery](./0038-database-access-aws-discovery.md)
+- [RFD 57: AWS EC2 nodes auto-discovery](https://github.com/gravitational/teleport/pull/12410)
+
+## Why
+
+The proposal is aimed at providing an easy way for Teleport administrators to
+connect their cloud-hosted resources (EC2 instances, RDS databases, EKS clusters
+an so on) to a Teleport cluster.
+
+Over the past few releases Teleport has been adding automatic discovery
+capabilities allowing it to find and register AWS databases, EC2 instances
+and (WIP as of this writing) EKS clusters. Despite providing an improved UX
+compared to registering the resources manually, connecting resources to cluster
+and setting up auto-discovery remains cumbersome with multiple different
+`teleport configure` commands and does not provide good visibility into the
+discovered resources.
+
+The `teleport discover` approach aims to take advantage of the existing auto
+discovery mechanisms Teleport has and provide a unified approach for users to
+enroll their cloud resources.
+
+## Scope
+
+- Works with both self-hosted edition and Teleport Cloud.
+- Focuses on discovering resources in AWS, other clouds to follow.
+- Covers currently supported cloud resources: EC2 instances and RDS databases.
+- The phase 1 of the implementation will only support EC2 instances and SSH.
```

Comment:
The phase 2 will support RDS as well?

Reply:
Yes, the idea is to split implementation in phases and implement the discover flow for EC2 instances first, then RDS and other databases. I'll make implementation phases more explicit in the RFD.